tests/test_utils.py

```diff
@@ -832,3 +927,69 @@ def test_validate_email_address(email_address, is_valid):
 )
 def test_shorten_str(original, shorten_by, shortened):
     assert shorten_str(original, shorten_by) == shortened
+
+
+@pytest.mark.asyncio
+async def test_get_event_loop_uvloop():
+    with patch("asyncio.set_event_loop_policy") as set_event_loop_policy_mock:
+        get_event_loop(uvloop=True)
+        set_event_loop_policy_mock.assert_called()
+
+
+@pytest.mark.asyncio
+async def test_get_event_loop_uvloop_when_exception():
+    with patch("asyncio.set_event_loop_policy", side_effect=Exception("welp")):
+        # Event if there's an exception, the loop is created
+        get_event_loop(uvloop=True)
+        assert True
+
+
+@pytest.mark.asyncio
+async def test_get_event_loop_uvloop_when_runtime_exception():
+    with patch("asyncio.get_running_loop", side_effect=RuntimeError("welp")):
+        # If loop is not created beforehands, the test will fail randomly
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+
+        returned_loop = get_event_loop()
+
+        assert loop == returned_loop
+        returned_loop.close()
+        loop.close()
+
+
+@pytest.mark.asyncio
+async def test_get_event_loop_uvloop_when_runtime_exception_and_loop_policy_has_no_loop():
+    event_loop_policy_mock = Mock()
+    event_loop_policy_mock.get_event_loop = Mock(return_value=None)
+    with patch("asyncio.get_running_loop", side_effect=RuntimeError("welp")), patch(
+        "asyncio.get_event_loop_policy", return_value=event_loop_policy_mock
+    ):
+        loop = get_event_loop()
+
+        assert True
+
+        loop.close()
+
+
+@pytest.mark.parametrize(
+    "strategy, interval, retry, expected_sleep",
+    [
+        (RetryStrategy.CONSTANT, 10, 2, 10),  # Constant 10 seconds
+        (RetryStrategy.LINEAR_BACKOFF, 10, 0, 0),  # 10 * 0 = 0
+        (RetryStrategy.LINEAR_BACKOFF, 10, 1, 10),  # 10 * 1 = 10
+        (RetryStrategy.LINEAR_BACKOFF, 10, 2, 20),  # 10 * 2 = 20
+        (RetryStrategy.EXPONENTIAL_BACKOFF, 10, 0, 1),  # 10 ^ 0 = 1
```

Comment:
is `0` a valid value for a retry counter? Just checking that we're not doing a `sleep(1)` before our first try for every request if we choose an exponential strategy.